Lesson23/app/Busra/script.js

```diff
@@ -0,0 +1,134 @@
+const formElement = document.querySelector('form');
+const searchInput = document.getElementById('search-input');
+const searchResults = document.querySelector('.search-results');
+const showMoreButton = document.getElementById('show-more-button');
+
+let inputData = '';
+let page = 1;
+
+async function searchImages() {
+  inputData = searchInput.value;
+  const url = `https://api.unsplash.com/search/photos?page=${page}&query=${inputData}&client_id=${accessKey}`;
+
+  const response = await fetch(url);
+  const data = await response.json();
+
+  const results = data.results;
+  if (page === 1) {
+    searchResults.innerHTML = '';
+  }
+
+  results.map((result) => {
+    const imageContainer = document.createElement('div');
+    imageContainer.classList.add('search-result');
+    const image = document.createElement('img');
+    image.src = result.urls.small;
+    image.alt = result.alt_description;
+
+    const imageLink = document.createElement('a');
+    imageLink.href = result.links.html;
+    imageLink.target = '_blank';
+    imageLink.textContent = result.alt_description;
+
+    const photographer = document.createElement('p');
+    const photographerLink = document.createElement('a');
+    photographerLink.href = result.user.links.html;
+    photographerLink.target = '_blank';
+    photographerLink.textContent = `📸 ${result.user.name}`;
+    photographer.appendChild(photographerLink);
+
+    const likeButton = document.createElement('button');
+    likeButton.textContent = '🤍';
+    likeButton.classList.add('like-button');
+
+    const likedImages = JSON.parse(localStorage.getItem('likedImages')) || [];
+
+    if (likedImages.includes(result.id)) {
+      likeButton.classList.add('liked');
+    } else {
+      likeButton.classList.remove('liked');
+    }
+
+    function toggleLike(result, likeButton) {
+      let likedImages = JSON.parse(localStorage.getItem('likedImages')) || [];
+
+      if (likedImages.includes(result.id)) {
+        const index = likedImages.indexOf(result.id);
+        likedImages.splice(index, 1);
+        likeButton.textContent = '🤍';
+        likeButton.classList.remove('liked');
+      } else {
+        likedImages.push(result.id);
+        likeButton.classList.add('liked');
+        likeButton.textContent = '🖤';
+      }
+
+      localStorage.setItem('likedImages', JSON.stringify(likedImages));
+    }
+
+    likeButton.addEventListener('click', () => {
+      toggleLike(result, likeButton);
+    });
+
+    const likeButtonHtml = document.querySelectorAll('like-button');
+
+    // fullscreende olduğu gibi default olarak ekranda bulunan like butonuna tıklandığında togglelike fonksiyonunu çalıştırmak istiyorum fakat olmuyor consolda herhangi bir hata da almıyorum
+
+    likeButtonHtml.forEach((button) => {
+      button.addEventListener('click', toggleLike);
```

Comment:
The code here does not work because toggleLike function expects result, likeButton as parameters but they are not provided when adding the event listener for all buttons. 

toggleLike function does not actually need entire result, it only needs image id. So you can replace the parameter in the function first, then add id for the hardcoded images by e.g. using data* attribute: https://www.w3schools.com/tags/att_data-.asp#:~:text=The%20data%2D*%20attribute%20gives,server%2Dside%20database%20queries).